core/models/channel.py

```diff
@@ -32,3 +32,18 @@ class Meta:
 
     def __str__(self):
         return f"{self.name} [{self.status}]"
+    
+class GameChannelMember(models.Model):
+    """Object that defines membership in a channel"""
+
+    read_messages = models.BooleanField(null=False, default=True)
+    send_messages = models.BooleanField(null=False, default=True)
+    read_message_history = models.BooleanField(null=False, default=True)
+    use_slash_commands = models.BooleanField(null=False, default=True)
+    manage_messages = models.BooleanField(null=False, default=False)
+
+    channel = models.ForeignKey(GameChannel, on_delete=models.CASCADE, related_name="text_channel")
+    user = models.ForeignKey(CustomUser, on_delete=models.CASCADE, related_name="text_channel")
```

Comment:
Related names define the reverse query - these names are going to be a little confusing if you're trying to get back to this model from say a user or a channel